include/openssl/e_os2.h

```diff
@@ -138,6 +138,44 @@ extern "C" {
 #  endif
 # endif
 
+/* ---------------------------- HP NONSTOP -------------------------------- */
+# ifdef __TANDEM
+#  ifdef NO_GETPID
+inline int nssgetpid();
```

Comment:
This is a public header.. Are you sure you want it here?

Reply:
The `nssgetpid()` method is a redirect of `getpid()` across the board. The platform port actually runs in a non-POSIX environment as well as the POSIX flavour. This particular code has been around a while in the port. I would rather this was elsewhere too, but `getpid()`, in fact `pid_t` also, is not available for that variant. The use of `getpid()` is rather pervasive so I think the choice was to do it here rather than everywhere to minimize the code impact. Is there a better place for this and the follow-on definitions?

Reply:
The point is that you have the exact same code in `e_os.h` and `e_os2.h`, which seems unnecessary.

If there are things that are for internal OpenSSL use, then that should be in `e_os.h` *only*.
If there are things meant to be for public consumption, i.e. something affect our public headers, then that should be in `e_os2.h` *only*
(if some goes in one and some in the other, you'll get to split some of the code apart)

Reply:
I am pretty sure (maybe 60%-ish) that the `e_os2.h` code is referenced in a calling application wrapper in one situation - not part of the port but used by it. I don't have access to that application, so I'm reluctant to move the code out of public.

The config and related changes are going to take a while to get right and I'm not going to be able to get there until after Sept 16th (although I'll try).

Reply:
Why should openssl carry stuff publically that concerns a completely different source set? That tells me that the application you're talking about should handle its own and that the code added in `e_os2.h` isn't needed for openssl needs, neither public nor private.

Reply:
> 
> 
> Why should openssl carry stuff publically that concerns a completely different source set? That tells me that the application you're talking about should handle its own and that the code added in `e_os2.h` isn't needed for openssl needs, neither public nor private.

Briefly - I could go into much more detail in private - there are two personalities on NonStop: GUARDIAN (legacy) and OSS (POSIX-like). I have been maintaining the POSIX-side port of OpenSSL since early 2015 (from 1.0.2-ish). The GUARDIAN side, where there are no pids - you would see `cpu` and `pin` in the `nssgetpid()` code - has a very different interface layer and wrapper to run there. The `nssgetpid()` call is a bridge to allow SSL processes within the GUARDIAN environment to be managed, so it needs to be externalized for that environment.

Reply:
It's still just for OpenSSL, and as far as I can tell, this shouldn't be needed in `e_os2.h` at all, the stuff in `e_os.h` should be enough.

If there are OpenSSL files that don't compile because of a missing `getpid()`, they should include `"e_os.h"` to get the proper definition.

Reply:
Should be done in commit  2716b338b9.